### PR TITLE
Use double VectorTableLookup on ARM in ProbabilisticMap

### PR DESCRIPTION
**Pull request by MihaZupan**
Contributes to #84328

Before
|         Method |     Mean |     Error |
|--------------- |---------:|----------:|
| IndexOfAnyProb | 3.776 µs | 0.0024 µs |

After
|         Method |     Mean |     Error |
|--------------- |---------:|----------:|
| IndexOfAnyProb | 2.576 µs | 0.0011 µs |

### CODE REVIEW

**Comment by danmoseley** [comment]
> the >>> operator, which apparently isn't turned into AdvSimd.ShiftRightLogical. 

Bug?

**Comment by MihaZupan** [comment]
Likely just not implemented yet, #75770 may be the right tracking issue for that.

**Comment by a74nh** [comment]
Code looks good to me. Out of curiosity, do you know roughly how much of the performance gain comes from the `VectorTableLookup` and how much comes from the `ShiftRightLogical` ?

**Comment by tannergooding** [comment]
> but later regressed it with https://github.com/dotnet/runtime/commit/22953700f9e0976e4a00e34ccf69a19cf346e1a1 by using the >>> operator, which apparently isn't turned into AdvSimd.ShiftRightLogical

Can you log a bug for this. Looks like a simple fix as we should just need to recognize the operators in addition to the named methods: https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/hwintrinsicarm64.cpp#L1520-L1555

**Comment by MihaZupan** [comment]
> how much of the performance gain comes from the VectorTableLookup

The 3.776 µs => 2.576 µs change on this benchmark is entirely due to `VectorTableLookup`.

> Can you log a bug for this.

#85257

File: src/libraries/System.Private.CoreLib/src/System/IndexOfAnyValues/ProbabilisticMap.cs [hunk 1]
```diff
@@ -167,15 +167,24 @@ private static Vector128<byte> IsCharBitSet(Vector128<byte> charMapLower, Vector
             // X86 doesn't have a logical right shift intrinsic for bytes: https://github.com/dotnet/runtime/issues/82564
             Vector128<byte> highNibble = Sse2.IsSupported
                 ? (values.AsInt32() >>> VectorizedIndexShift).AsByte() & Vector128.Create((byte)15)
-                : values >>> VectorizedIndexShift;
+                : AdvSimd.ShiftRightLogical(values, VectorizedIndexShift);
```
Comment (MihaZupan) [inline src/libraries/System.Private.CoreLib/src/System/IndexOfAnyValues/ProbabilisticMap.cs:170]:
```suggestion
                : values >>> VectorizedIndexShift;
```

**Comment by kunalspathak** [comment]
> The 3.776 µs => 2.576 µs change on this benchmark is entirely due to `VectorTableLookup`.

That's 30% gain. Good to see that.

**Review by kunalspathak — APPROVED** [review]
LGTM

**Comment by MihaZupan** [comment]
Failures are known according to Build Analysis

**Pull request merged**